File: Easy/2026-06-02-3633-Earliest-Finish-Time-for-Land-and-Water-Rides-I/solution.java
class Solution {
    public int earliestFinishTime(int[] landStartTime, int[] landDuration, int[] waterStartTime, int[] waterDuration) {
        int earliestFinish = Integer.MAX_VALUE;

        for (int i = 0; i < landStartTime.length; i++) {
            for (int j = 0; j < waterStartTime.length; j++) {
                // Plan A: Land ride first, then water ride
                int landFinish = Math.max(landStartTime[i], 0) + landDuration[i];
                int waterStartAfterLand = Math.max(waterStartTime[j], landFinish);
                int waterFinish = waterStartAfterLand + waterDuration[j];
                earliestFinish = Math.min(earliestFinish, waterFinish);

                // Plan B: Water ride first, then land ride
                int waterFinishFirst = Math.max(waterStartTime[j], 0) + waterDuration[j];
                int landStartAfterWater = Math.max(landStartTime[i], waterFinishFirst);
                int landFinishAfterWater = landStartAfterWater + landDuration[i];
                earliestFinish = Math.min(earliestFinish, landFinishAfterWater);
            }
        }

        return earliestFinish;
    }
}

File: Easy/2026-06-02-3633-Earliest-Finish-Time-for-Land-and-Water-Rides-I/solution.js
function earliestFinishTime(landStartTime, landDuration, waterStartTime, waterDuration) {
    let minFinishTime = Infinity;

    for (let i = 0; i < landStartTime.length; i++) {
        for (let j = 0; j < waterStartTime.length; j++) {
            // Plan A: Land ride first, then water ride
            const landFinish = Math.max(landStartTime[i], 0) + landDuration[i];
            const waterStartAfterLand = Math.max(waterStartTime[j], landFinish);
            const waterFinish = waterStartAfterLand + waterDuration[j];
            minFinishTime = Math.min(minFinishTime, waterFinish);

            // Plan B: Water ride first, then land ride
            const waterFinishFirst = Math.max(waterStartTime[j], 0) + waterDuration[j];
            const landStartAfterWater = Math.max(landStartTime[i], waterFinishFirst);
            const landFinishAfterWater = landStartAfterWater + landDuration[i];
            minFinishTime = Math.min(minFinishTime, landFinishAfterWater);
        }
    }

    return minFinishTime;
}